### DISCUSSION TITLE
Issue building with tsup inside only on one package

### DISCUSSION THREAD

**Discussion by elrumordelaluz** [vercel/turborepo#1797] [Help] [answered]
I am enjoying a lot a recent switch I made from a [monorepo ](https://github.com/elrumordelaluz/reactour/tree/v2) with each package using `tsdx` into `turborepo` + `changesets`, but just after the first publications I found some issues, mirrored with issues opened in the public repository, asking to workarounds.

At a first glance I use the [basic demo](https://github.com/vercel/turborepo/tree/main/examples/basic) but then noticed that to build each package right now is needed to use your own approach, so following the other examples I started using `tsup`.

The problem I am having seems to be only on one package, described [here](https://github.com/vercel/turborepo/tree/main/examples/basic). The `mjs` output for this specific one seems compiled to commonjs and generates what I am supposing is a related issue, showing the message below when launching the `apps/web` demo.

> Error: Dynamic require of "react" is not supported

Is there a way to add an `/example` folder using some other bundler/transpiler to obtain the same result of the other examples?

I tried using `tsdx` and also `tsc` + `rollup` for this specific package, but for some reason, this created another side effect which is based on imports from one package to this one:

> Can't import the named export 'getPadding' from non EcmaScript module (only default export is available)

An example could be awesome but also any idea on how to solve in any way is pretty welcome!

Also thanks for this awesome project!

**Comment by anthonyshew** [comment, marked as answer]
Example for publishing: https://github.com/vercel/turbo/tree/main/examples/with-changesets

Example using a bundler (Rollup): https://github.com/vercel/turbo/tree/main/examples/with-rollup
